Source/WebInspectorUI/UserInterface/Base/BrowserInspectorFrontendHost.js

```diff
@@ -226,13 +226,25 @@ if (!window.InspectorFrontendHost) {
 
         canSave(saveMode)
         {
-            return false;
+            // Support single file mode only.
+            return saveMode === WI.FileUtilities.SaveMode.SingleFile;
         }
 
         save(saveDatas, forceSaveAs)
         {
-            // FIXME: Create a Blob from the content, get an object URL, open it to trigger a download.
-            throw "unimplemented";
+            console.assert(saveDatas.length === 1, saveDatas);
+
+            let saveData = saveDatas[0];
+            let blob = WI.BlobUtilities.blobForContent(saveData.content, saveData.base64Encoded, "octet/stream");
```

Comment:
why `"octet/stream"`?

could/should we have that be a fallback for if something like `WI.mimeTypeForFileExtension(saveData.url)` doesn't work?

Reply:
"octet-stream" was a typo. Should have been "application/octet-stream". It's a fallback in case the download attribute of the <a> element doesn't work. It ensures that the browser will download the link instead of trying to open it, despite it may not be necessary in recent browsers.
I've changed it so we use WI.mimeTypeForFileExtension(saveData.url) as you suggested, and fallback to "application/octet-stream" if that function can't guess the mime type of the file (which happens for .har files).